python/utils/generatejson.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/python
+
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This script generates request json for the Vision API.
+You can run this script like this:
+python generatejson.py \
+    -i <inputfile> \
+    -o <outputfile.json>
+where the input file contains info on how to process a set of URIs, and the
+output is a json request file.
+
+Each line of the input file contains a URI (e.g. a file path), and one or more
+numbered feature:maxResults specifications. The feature numbers are enumerated
+in the 'getDetectionType' function.
+E.g.:
+/Users/username/<username>/desktv.jpg 2:10 3:10 4:10
+See https://cloud.google.com/vision/docs/concepts for more detail.
+
+Then, see https://cloud.google.com/vision/docs/concepts for information on
+using the generated result to send a request to the Vision API.
+"""
+
+import base64
+import collections
+import getopt
+import json
+import sys
+
+
+def main(argv):
+    inputfile = ''
+    outputfile = ''
+    try:
+        opts, args = getopt.getopt(argv, "hi:o:", ["ifile=", "ofile="])
+    except getopt.GetoptError:
+        print 'generatejson.py -i <i>inputfile</i> -o <i>outputfile</i>'
+        sys.exit(2)
+    for opt, arg in opts:
+        if opt == '-h':
+            print 'generatejson.py -i <i>inputfile</i> -o <i>outputfile</i>'
+            print 'Input File format:'
+            print 'FileURI Feature:maxResults ...'
+            sys.exit()
+        elif opt in ("-i", "--ifile"):
+            inputfilename = arg
+        elif opt in ("-o", "--ofile"):
+            outputfilename = arg
```

Comment:
FYI what we've been doing for other python samples is to put the argument parsing into the `if __name__` block, and using the argparse lib, which is nicer and cleaner than getopt. Check out [list_objects.py](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/storage/api/list_objects.py#L68), for instance.